src/java/com/zimbra/oauth/utilities/OAuthDataSource.java

```diff
@@ -129,7 +129,16 @@ public void syncDatasource(ZMailbox mailbox, OAuthInfo credentials) throws Servi
                 // ensure the specified storage folder exists
                 final String folderName = String
                     .format(OAuth2Constants.DEFAULT_OAUTH_FOLDER_TEMPLATE, username, client);
-                final String storageFolderId = ensureFolder(mailbox, folderName, View.contact);
+                View view = null;
+                if (dsType == DataSourceType.oauth2contact) {
+                    view = View.contact;
+                } else if (dsType == DataSourceType.oauth2calendar) {
+                    view = View.appointment;
+                } else {
+                    ZimbraLog.extensions.errorQuietly("Invalid data source type received", null);
```

Comment:
We use this only when we want the stack trace to be printed if debug is enabled. Since e is null, we can just use  ZimbraLog.extensions.error()

Reply:
done